### PR TITLE
[SRE-202] Fix stylish-haskell and nixfmt

### PR DESCRIPTION
**Pull request by craigem**
Additionally remove all references to buildkite.

### CODE REVIEW

**Comment by EncodePanda** [comment]
@craigem seems we are missing just the removal of the buildkite from github checks, or am I missing something?

**Comment by craigem** [comment]
`nixfmt` tests work at this point.

Circling back to `stylish-haskell`.

File: nix/network-docs.nix [hunk 1]
```diff
@@ -1,34 +1,25 @@
...
-        cleveref
-        framed
-        scheme-small
-        collection-fontsrecommended
-        stmaryrd kpfonts geometry hyperref
-        todonotes
-        amsmath mathtools
-        colortbl polytable lazylist
-        fancyvrb
+        cleveref framed scheme-small collection-fontsrecommended stmaryrd
```
Comment (dnadales) [inline nix/network-docs.nix:16]: Personally, I find it more readable to have one package per-line. Also, it makes it easy to comment out something when tinkering with a nix setup.

File: nix/stack-shell.nix [hunk 2]
```diff
@@ -2,6 +2,13 @@ with import ./. { };
 
 haskell.lib.buildStackProject {
   name = "stack-env";
-  buildInputs = with pkgs; [ zlib openssl gmp libffi git haskellPackages.happy ];
+  buildInputs = with pkgs; [
+    zlib
```
Comment (dnadales) [inline nix/stack-shell.nix:6]: So for instance here we're using a multi-line packages specification. Maybe it's a good idea to stick to a single style.

File: ouroboros-consensus-shelley-test/test/Test/ThreadNet/Shelley.hs [hunk 3]
```diff
@@ -114,15 +114,6 @@ instance Arbitrary NightlyTestSetup where
   arbitrary = do
       setup <- arbitrary
 
-      -- At time of writing, this causes 100 tests to have an expected run time
```
Comment (dnadales) [inline ouroboros-consensus-shelley-test/test/Test/ThreadNet/Shelley.hs:117]: I'm not sure this comment is no longer relevant. It explains why we chose the frequency numbers. @nfrisby @nc6 thoughts?

Comment (nfrisby) [reply ouroboros-consensus-shelley-test/test/Test/ThreadNet/Shelley.hs:117]: Yep, I think this information would help future time tuning. @craigem would you explain why the removal? Thanks

File: nix/stack-shell.nix (hunk 2 above)
Comment (craigem) [reply nix/stack-shell.nix:6]: This formatting has been done automatically by `nixfmt`.

File: ouroboros-consensus-shelley-test/test/Test/ThreadNet/Shelley.hs (hunk 3 above)
Comment (craigem) [reply ouroboros-consensus-shelley-test/test/Test/ThreadNet/Shelley.hs:117]: This was removed due to referencing buildkite which this PR removes,

Happy to add it back in.

**Comment by craigem** [comment]
A rebase on master added:

```
scripts/ci/check-stylish-network.sh
```
Do we want this to be a Hydra job as part of this PR too?

Otherwise, Hydra pending, this PR is pretty much ready.

File: ouroboros-consensus-shelley-test/test/Test/ThreadNet/Shelley.hs (hunk 3 above)
Comment (dnadales) [reply ouroboros-consensus-shelley-test/test/Test/ThreadNet/Shelley.hs:117]: Yeah, I think it's wise to leave this comment. If we want to remove the reference to Buildkite maybe we can rephrase this to something like:

```
of half an hour on a CI machine.
```

but I don't know if that's any better.

Comment (craigem) [reply ouroboros-consensus-shelley-test/test/Test/ThreadNet/Shelley.hs:117]: Will do that push it back.

Comment (craigem) [reply ouroboros-consensus-shelley-test/test/Test/ThreadNet/Shelley.hs:117]: Re-added in the most recent commit.

Comment (dnadales) [reply ouroboros-consensus-shelley-test/test/Test/ThreadNet/Shelley.hs:117]: Thanks Craige. I'm just afraid that "the queue=bench machine" is meaningless. That's why I suggested referring to a CI machine, or maybe keeping the reference to buildkite:

```
This caused 100 tests to have an expected run time of half an hour in a Buildkite machine. Note that the Buildkite CI infrastructure is now deprecated [in favour of ...]. 

```

or something like that.

Comment (craigem) [reply ouroboros-consensus-shelley-test/test/Test/ThreadNet/Shelley.hs:117]: I hope to have address this in the most recent push.

Comment (dnadales) [reply ouroboros-consensus-shelley-test/test/Test/ThreadNet/Shelley.hs:117]: Looks good. Thank you Craige :pray:

**Review by dnadales — APPROVED** [review]

**Review by disassembler — APPROVED** [review]

**Comment by craigem** [comment]
bors merge

**Comment by craigem** [comment]
Due to this commit removing the buildkite CI which bors is failing on, we may need someone to manually merge this PR.

**Comment by EncodePanda** [comment]
this change breaks direnv :(

**Comment by craigem** [comment]
> this change breaks direnv :(

There's no default `.envrc` in this repo.

What's in your .envrc (perhaps share via DM?).

**Comment by craigem** [comment]
> this change breaks direnv :(

Are you able to provide more detail around what is broken from your perspective?

A simple `.envrc` like this:

```
use nix
watch_file nix/*
```

works fine for me.

**Comment by EncodePanda** [comment]
My `.envrc` is just `use nix`. Which normally suffice for `direnv + nix-direnv`

**Comment by craigem** [comment]
I'll try an `.envrc` with just that.

In the mean time can you describe what you're experiencing as broken and what should I expect to see when it's fixed?

**Comment by EncodePanda** [comment]
1. Entering nix-shell without issues

```
encodepanda in ~/projects/ouroboros-network on EncodePanda/CAD-3718-current-mempool ● ● λ nix-shell
trace: Using haskell.nix's nixpkgs. Good. Sharing is caring
trace: WARNING: 8.10.4 is out of date, consider using 8.10.7.
building '/nix/store/fddqnjvcpd7chg7mbjdmclwwhm0ymdss-git-ls-files.drv'...
trace: To make project.plan-nix for haskell-project a fixed-output derivation but not materialized, set `plan-sha256` to the output of the 'calculateMaterializedSha' script in 'passthru'.
trace: To materialize project.plan-nix for haskell-project entirely, pass a writable path as the `materialized` argument and run the 'updateMaterialized' script in 'passthru'.
trace: WARNING: 8.10.4 is out of date, consider using 8.10.7.
trace: To make project.plan-nix for haskell-project a fixed-output derivation but not materialized, set `plan-sha256` to the output of the 'calculateMaterializedSha' script in 'passthru'.
trace: To materialize project.plan-nix for haskell-project entirely, pass a writable path as the `materialized` argument and run the 'updateMaterialized' script in 'passthru'.
trace: To make project.plan-nix for cabal-install a fixed-output derivation but not materialized, set `plan-sha256` to the output of the 'calculateMaterializedSha' script in 'passthru'.
trace: To materialize project.plan-nix for cabal-install entirely, pass a writable path as the `materialized` argument and run the 'updateMaterialized' script in 'passthru'.
trace: To make project.plan-nix for stylish-haskell a fixed-output derivation but not materialized, set `plan-sha256` to the output of the 'calculateMaterializedSha' script in 'passthru'.
trace: To materialize project.plan-nix for stylish-haskell entirely, pass a writable path as the `materialized` argument and run the 'updateMaterialized' script in 'passthru'.
trace: To make project.plan-nix for haskell-project a fixed-output derivation but not materialized, set `plan-sha256` to the output of the 'calculateMaterializedSha' script in 'passthru'.
trace: To materialize project.plan-nix for haskell-project entirely, pass a writable path as the `materialized` argument and run the 'updateMaterialized' script in 'passthru'.
trace: No index state specified for eventlog2html, using the latest index state that we know about (2021-11-26T00:00:00Z)!
trace: No index state specified for ghcid, using the latest index state that we know about (2021-11-26T00:00:00Z)!
trace: No index state specified for graphmod, using the latest index state that we know about (2021-11-26T00:00:00Z)!
trace: No index state specified for hasktags, using the latest index state that we know about (2021-11-26T00:00:00Z)!
trace: No index state specified for hp2pretty, using the latest index state that we know about (2021-11-26T00:00:00Z)!
trace: No index state specified for profiteur, using the latest index state that we know about (2021-11-26T00:00:00Z)!
```

2. cabal available in nix-shell

```
[nix-shell:~/projects/ouroboros-network]$ cabal --version
/nix/store/b2bjispinr1841b3wrmjsqsg2rlhz1wh-cabal-install-exe-cabal-3.6.2.0/bin/cabal --project-file=/Users/encodepanda/projects/ouroboros-network/.nix-shell-cabal.project --version
cabal-install version 3.6.2.0
compiled using version 3.6.2.0 of the Cabal library
```

3. after exiting nix-shell let's `direnv reload` just in case (this should not be necessary but we can do it just in case)

```
[nix-shell:~/projects/ouroboros-network]$ exit
exit
encodepanda in ~/projects/ouroboros-network on EncodePanda/CAD-3718-current-mempool ● ● λ direnv reload
direnv: loading ~/projects/ouroboros-network/.envrc
direnv: using nix
trace: Using haskell.nix's nixpkgs. Good. Sharing is caring
trace: WARNING: 8.10.4 is out of date, consider using 8.10.7.
trace: To make project.plan-nix for haskell-project a fixed-output derivation but not materialized, set `plan-sha256` to the output of the 'calculateMaterializedSha' script in 'passthru'.
trace: To materialize project.plan-nix for haskell-project entirely, pass a writable path as the `materialized` argument and run the 'updateMaterialized' script in 'passthru'.
trace: WARNING: 8.10.4 is out of date, consider using 8.10.7.
trace: To make project.plan-nix for haskell-project a fixed-output derivation but not materialized, set `plan-sha256` to the output of the 'calculateMaterializedSha' script in 'passthru'.
trace: To materialize project.plan-nix for haskell-project entirely, pass a writable path as the `materialized` argument and run the 'updateMaterialized' script in 'passthru'.

direnv: eval /Users/encodepanda/projects/ouroboros-network/.direnv/cache-pre282548.4c87cb87a2d
direnv: export +IN_NIX_SHELL +TEMPDIR +XDG_DATA_DIRS ~PATH
``` 

4. cabal not available

```
encodepanda in ~/projects/ouroboros-network on EncodePanda/CAD-3718-current-mempool ● ● λ cabal --version
zsh: command not found: cabal
```

5. let's exit the folder and re-enter it

```
encodepanda in ~/projects/ouroboros-network on EncodePanda/CAD-3718-current-mempool ● ● λ cd ..
direnv: unloading
encodepanda in ~/projects λ cd ouroboros-network
direnv: loading ~/projects/ouroboros-network/.envrc
direnv: using nix
direnv: using cached derivation
direnv: eval /Users/encodepanda/projects/ouroboros-network/.direnv/cache-pre282548.4c87cb87a2d
direnv: export +IN_NIX_SHELL +TEMPDIR +XDG_DATA_DIRS ~PATH
encodepanda in ~/projects/ouroboros-network on EncodePanda/CAD-3794/unify ● λ cabal --version
zsh: command not found: cabal
```

**Comment by EncodePanda** [comment]
Let me show you how that works on a different project of mine that is using `haskell.nix` (and how it used to work in `ouroboros-network` just very recently)

```
encodepanda in ~ λ cabal --version
zsh: command not found: cabal

encodepanda in ~ λ j purely
/Users/encodepanda/projects/purely-functional-data-structures
direnv: loading ~/projects/purely-functional-data-structures/.envrc
direnv: using nix
direnv: using cached derivation
direnv: eval /Users/encodepanda/projects/purely-functional-data-structures/.direnv/cache-pre282548.4c87cb87a2d
direnv: export +AR +AS +CC +CONFIG_SHELL +CXX +HOST_PATH +IN_NIX_SHELL +LANG +LD +LD_DYLD_PATH +LOCALE_ARCHIVE +MACOSX_DEPLOYMENT_TARGET +NIX_BINTOOLS +NIX_BINTOOLS_WRAPPER_x86_64_apple_darwin_TARGET_HOST +NIX_BUILD_CORES +NIX_BUILD_DONT_SET_RPATH +NIX_BUILD_TOP +NIX_CC +NIX_CC_WRAPPER_x86_64_apple_darwin_TARGET_HOST +NIX_CFLAGS_COMPILE +NIX_COREFOUNDATION_RPATH +NIX_CXXSTDLIB_COMPILE +NIX_CXXSTDLIB_LINK +NIX_DONT_SET_RPATH +NIX_ENFORCE_NO_NATIVE +NIX_GHC_LIBDIR +NIX_HARDENING_ENABLE +NIX_IGNORE_LD_THROUGH_GCC +NIX_INDENT_MAKE +NIX_LDFLAGS +NIX_NO_SELF_RPATH +NIX_STORE +NM +PATH_LOCALE +RANLIB +SDKROOT +SIZE +SOURCE_DATE_EPOCH +STRINGS +STRIP +TEMP +TEMPDIR +TMP +XDG_DATA_DIRS +__ETC_PROFILE_SOURCED +__darwinAllowLocalNetworking +__impureHostDeps +__propagatedImpureHostDeps +__propagatedSandboxProfile +__sandboxProfile +buildInputs +builder +configureFlags +depsBuildBuild +depsBuildBuildPropagated +depsBuildTarget +depsBuildTargetPropagated +depsHostHost +depsHostHostPropagated +depsTargetTarget +depsTargetTargetPropagated +doCheck +doInstallCheck +gl_cv_func_getcwd_abort_bug +installPhase +name +nativeBuildInputs +out +outputs +patches +phases +propagatedBuildInputs +propagatedNativeBuildInputs +shell +shellHook +stdenv +strictDeps +system ~PATH ~TMPDIR

encodepanda in ~/projects/purely-functional-data-structures on master ● ● λ cabal --version
cabal-install version 3.2.0.0
compiled using version 3.2.0.0 of the Cabal library

encodepanda in ~/projects/purely-functional-data-structures on master ● ● λ cat .envrc
use nix
```

**Comment by craigem** [comment]
I see:
```
ouroboros-network on  sre-203/addCheckStylishNetwork [$?] on ☁️  (eu-central-1) took 4s 
❯ cat .envrc               
use nix


ouroboros-network on  sre-203/addCheckStylishNetwork [$?] via ❄️  impure (cabal-dev-shell) on ☁️  (eu-central-1) took 43s 
❯ direnv reload
direnv: loading ~/source/IOG/ouroboros-network/.envrc                                                                                                                                                                                                                                                                          
direnv: using nix
trace: Using haskell.nix's nixpkgs. Good. Sharing is caring
...

ouroboros-network on  sre-203/addCheckStylishNetwork [$?] via ❄️  impure (cabal-dev-shell) on ☁️  (eu-central-1) took 18s 
❯ cabal --version
/nix/store/spaxwxlm2qmk8zq0bpvzppvg1m3jmh0v-cabal-install-exe-cabal-3.6.2.0/bin/cabal --project-file=/home/craige/source/IOG/ouroboros-network/.nix-shell-cabal.project --version
cabal-install version 3.6.2.0
compiled using version 3.6.2.0 of the Cabal library
```

I can't see why you're getting a different result :thinking:

**Comment by EncodePanda** [comment]
also my `.direnv` cache file is empty for some reason

```
encodepanda in ~/projects/ouroboros-network on EncodePanda/missing-golden ● ● λ ls .direnv/cache-pre282548.4c87cb87a2d
.direnv/cache-pre282548.4c87cb87a2d
encodepanda in ~/projects/ouroboros-network on EncodePanda/missing-golden ● ● λ cat .direnv/cache-pre282548.4c87cb87a2d
encodepanda in ~/projects/ouroboros-network on EncodePanda/missing-golden ● ● λ cd .direnv
encodepanda in ~/projects/ouroboros-network/.direnv on EncodePanda/missing-golden ● λ ls
cache-pre282548.4c87cb87a2d
encodepanda in ~/projects/ouroboros-network/.direnv on EncodePanda/missing-golden ● λ bat cache-pre282548.4c87cb87a2d
───────┬──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
       │ File: cache-pre282548.4c87cb87a2d   <EMPTY>
       ````

Usually it would be full of exports

**Comment by EncodePanda** [comment]
I've dropped the whole `.direnv` to remove all caches. But the problem remains.
